python/tests/unit/arcticdb/version_store/test_append.py

```diff
@@ -288,3 +288,34 @@ def test_(initial: InputFactories, append: InputFactories, match, swap, lmdb_ver
     to_append, _ = append.make(abs(next_start), 1)
     with pytest.raises(NormalizationException):
         lib.append("s", to_append, match=match)
+
+
+def test_append_non_sorted_exception(lmdb_version_store):
+    symbol = "bad_append"
+
+    num_initial_rows = 20
+    initial_timestamp = pd.Timestamp("2019-01-01")
+    dtidx = pd.date_range(initial_timestamp, periods=num_initial_rows)
+    df = pd.DataFrame(
+        {
+            "uint8": _random_integers(num_initial_rows, np.uint8),
+            "uint32": _random_integers(num_initial_rows, np.uint32),
+        },
+        index=dtidx,
+    )
+    assert df.index.is_monotonic_increasing == True
+
+    lmdb_version_store.write(symbol, df)
+    info = lmdb_version_store.get_info(symbol)
+    assert info["sorted"] == "ASCENDING"
+
+    num_rows = 20
+    initial_timestamp = pd.Timestamp("2020-01-01")
+    dtidx = np.roll(pd.date_range(initial_timestamp, periods=num_initial_rows), 3)
+    df2 = pd.DataFrame(
+        {"uint8": _random_integers(num_rows, np.uint8), "uint32": _random_integers(num_rows, np.uint32)}, index=dtidx
+    )
+    assert df2.index.is_monotonic_increasing == False
```

Comment:
The fixture setup is a bit too long, which makes it hard to see what the test is actually checking. (IMO, the code doesn't care about the types of the columns nor the number of rows nor randomness.)

Please consider using existing fixture creation functions.